src/Compilers/CSharp/Test/Emit2/Semantics/InlineArrayTests.cs

```diff
@@ -23106,5 +23106,35 @@ struct ThreeStringBuffer {
             var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80, options: TestOptions.ReleaseExe);
             var verifier = CompileAndVerify(comp, expectedOutput: "123124").VerifyDiagnostics();
         }
+
+        [Fact]
+        public void Initialization_RefStruct_Await()
+        {
+            var src = """
+                using System.Threading.Tasks;
+
+                var b = new Buffer();
+                b[0] = await GetInt();
+                b[1] = await GetInt();
+
+                static Task<int> GetInt() => Task.FromResult(42);
+                
+                [System.Runtime.CompilerServices.InlineArray(4)]
+                ref struct Buffer
```

Comment:
>ref struct Buffer [](http://example.com/codeflow?start=16&length=17)

I would be good to have a test when this is not a ref struct. The scenario will use references underneath.